### DISCUSSION TITLE
List fields in the model

### DISCUSSION THREAD

**Discussion by andim27** [nylo-core/nylo#52] [Ideas] [answered]
Can we store List<T> field to the Model and store it to NyStorage(Not collection) ? Or may be List<Map<String,String>> for the next release?

**Comment by agordn52** [comment, marked as answer]
Hi @andim27,

Have you got an example?

At the moment, you can store `Model`'s to a collection like:
``` dart
User userOne = User();
userOne.addToCollection('key');

User userTwo = User();
userTwo.addToCollection('key');

await NyStorage.readCollection('key'); // [User, User]
```

Thanks,

**Comment by andim27** [comment]
No, I mean fields of User. Now it can be String, int, double. Can we add List type?

**Reply by agordn52** [reply]
I'm struggling to understand this, can you give an example?

**Comment by andim27** [comment]
When the following is scheduled to be released?
`User user = await NyStorage.read<User>('auth_user', model: new User())`
It seems not working in ver 5.1

**Reply by agordn52** [reply]
@andim27, in v5.1.0+ you can do `User user = await NyStorage.read<User>('auth_user')` and that'll work.
Just make sure that your Model is set in your `config/decoders.dart` file and your Model extends the `Model` class.

``` dart
/// Usage
/// class User extends Model {
///   String? name;
///   String? email;
///   User();
///   User.fromJson(dynamic data) {
///     name = data['name'];
///     email = data['email'];
///   }
///   toJson() => {
///     "name": name,
///     "email": email
///   };
/// }
```
